src/harness/fourslash.ts

```diff
@@ -1816,8 +1824,13 @@ namespace FourSlash {
                     }
                 }
                 runningOffset += editDelta;
-                // TODO: Consider doing this at least some of the time for higher fidelity. Currently causes a failure (bug 707150)
-                // this.languageService.getScriptLexicalStructure(fileName);
+
+                // Update positions of any future edits affected by this change
+                for (let j = i + 1; j < edits.length; j++) {
+                    if (edits[j].span.start >= edits[i].span.start) {
+                        edits[j].span.start += editDelta;
+                    }
+                }

```

Comment:
Given that this is test code, I don't particularly care about the extra work, but having the assert seems worthwhile.

---
In reply to: [132300325](https://github.com/Microsoft/TypeScript/pull/17625#discussion_r132300325) [](ancestors = 132300325)